framework/TSA/LinearRegression.py

```diff
@@ -0,0 +1,128 @@
+# Copyright 2017 Battelle Energy Alliance, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+  Linear and Polynomial Regression
+"""
+import numpy as np
+import utils.importerUtils
+statsmodels = utils.importerUtils.importModuleLazy("statsmodels", globals())
+
+from utils import InputData, InputTypes, randomUtils, xmlUtils, mathUtils, utils
+from .TimeSeriesAnalyzer import TimeSeriesAnalyzer
+
+
+class LinearRegression(TimeSeriesAnalyzer):
```

Comment:
aren't we replicating the LinearRegressor in RAVEN Surrogate models with the simple addition of the Polynomial bindings? Are you taking care of removing the redundancy from the Surrogate models?

Reply:
I looked for a polynomial regressor in RAVEN and couldn't find it. Where is it?

Reply:
I did some digging; the only linear regressor in RAVEN I could find is the ScitKitLearn SubType, which has several differences from what is in this PR:
- The SKL ROM is not specific to time series; it deals with perturbed input mapping to response surfaces
- The SKL ROM is limited to linear fits
- The SKL LinearRegressor is embedded as a subtype of the SKL ROM, making its availability limited
- The TSA PolyRegressor here is focused on exposing the characteristics of the fit, not on generating an engine for interpolative sampling

I can see renaming this from LinearRegressor to PolynomialRegressor to avoid the confusion, but I don't see any code duplication and very little feature duplication here. That's just my take.